webapp/src/components/CompetenciesPage.tsx

```diff
@@ -28,15 +28,15 @@ const CompetenciesPage = () => {
         <Grid item xs={12} md={8}>
           <Box>
             <CreateCompetencyForm
-              onCompetencyCreated={id =>
-                setNewlyCreatedCompetencyIds([...newlyCreatedCompetencyIds, id])
-              }
+              onCompetenciesChanged={id =>
+                setChangedCompetencyIds([...changedCompetencyIds, id])}
             />
           </Box>
           <Box my={6}>
             {competencies.length === 0 && <p>There are no competencies.</p>}
-            {competencies.map(({ id, label, description }) => (
-              <Competency key={id} description={description} label={label} />
+            {competencies.map(({ id, label, description, principal_id }) => (
+              <Competency key={id} id={id} description={description} label={label} principal_id={principal_id} onCompetenciesChanged={id =>
```

Comment:
Props in React should be camel case. `principal_id` should be `principalId` in that case.